hs_file_types/views.py

```diff
@@ -1201,12 +1201,11 @@ def get_metadata(request, hs_file_type, file_type_id, metadata_mode):
 
     from hs_core.views.utils import authorize
     from rest_framework.exceptions import PermissionDenied
-    if not logical_file.resource.raccess.public:
-        if request.user.is_authenticated:
-            authorize(request, logical_file.resource.short_id,
-                      needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
-        else:
-            raise PermissionDenied()
+
+    _, authorized, _ = authorize(request, logical_file.resource.short_id,
+                                 needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE, raises_exception=False)
+    if not authorized:
+        raise PermissionDenied()
```

Comment:
If raises_exception argument in authorize() is set to True, PermissionDenied() exception will be raised, so lines 1207-1208 will not be needed.

Reply:
The `PermissionDenied` raised here is an exception class from django rest framework, The one raised by the `authorize` is an exception class from django. I am not sure why we used originally this exception class of django rest framework here. I will make the change and see if it breaks any tests.